compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -278,6 +278,9 @@ class Namer { typer: Typer =>
                 if rhs.isEmpty || flags.is(Opaque) then flags |= Deferred
             if flags.is(Param) then tree.rhs else analyzeRHS(tree.rhs)
 
+        def isNonInferingTree(tree: ValOrDefDef): Boolean =
```

Comment:
Same somment as above for the spelling. I am also wondering if "hasExplicitType" wouldn't be clearer.